### ISSUE TITLE
versions after 4.4.6 break SlashRequirePermissions

### ISSUE THREAD

**Issue by Zbizu** [DSharpPlus/DSharpPlus#1979]
### Summary

The following code works in 4.4.6, but I get "application is not responding" when trying it with 4.4.7 and 4.4.8. Using NuGet in Visual Studio.
```cs
DiscordClient? client = Discord.GetInstance();
Discord.Commands = client.UseSlashCommands();
foreach (KeyValuePair<ulong, DiscordServerConfig> serverCfg in config.GetDiscordServers()) {
    InternalRegisterCommands(config, serverCfg.Value, Discord.Commands);
}
```

part of `InternalRegisterCommands`:
```cs
commands.RegisterCommands<TestCommands>(server.id);
```

test command:
```cs
public class TestCommands : ApplicationCommandModule {
    // command: /ping
    [SlashCommand("ping", "Sends a response in chat.")]
    [SlashRequirePermissions(Permissions.Administrator)]
    public async Task PingCommand(InteractionContext context) {

        // create message response
        await Response.ReplyToUser(context, "Pong! :ping_pong:");
    }
}
```

### What version of the library are you using?

v4.4.6 (Stable)

### What .NET version are you using? Make sure to use the latest patch release for your major version.

.NET 8.0

### Operating System

Ubuntu 22.04.3 LTS Jammy (arch64/arm)

### Reproduction Steps

1. create test slashcommand on 4.4.6
2. add `[SlashRequirePermissions(Permissions.Administrator)]` to it
3. test if it works
4. change version to 4.4.7 or 4.4.8
5. test if it works again

**Comment by Plerx2493** [comment]
The `CommandErrored` event on the extension should give you an exception in this case. If you dont get anything there, you should get an exception in the `ClientErrored` event on the `DiscordClient`

**Comment by Zbizu** [comment]
@Plerx2493 
Testing on Windows 10 (x64) now with Visual Studio debugger attached:
I'm not getting any exceptions at all, it just doesn't respond (first use is on 4.4.6)
![obraz](https://github.com/DSharpPlus/DSharpPlus/assets/5205079/ebbbe234-d18a-4041-b80d-c7c9c114bc6b)

When I compile in debug with `MinimumLogLevel = LogLevel.Debug`, nothing prints when I execute the command either

Commands without `SlashRequirePermissions` work fine on all versions:
![obraz](https://github.com/DSharpPlus/DSharpPlus/assets/5205079/cc148c0b-eafd-4483-95a5-2067a6ce8d05)

**Comment by Plerx2493** [comment]
Did you subscribe to the mentioned events? The debugger only shows uncatched exception afaik

**Comment by Zbizu** [comment]
@Plerx2493 I'm new to C# and this library, could you show me how to subscribe and to the event I need?

**Comment by Plerx2493** [comment]
https://dsharpplus.github.io/DSharpPlus/articles/beyond_basics/events.html

**Comment by Plerx2493** [comment]
This is clearly a bug and we should fix it, but I also recommend you to use our nightly version (prerelease version) to use all new discord features and spare you from migrating later. At least if I understand stand you correctly and you just started to use this library

**Comment by Zbizu** [comment]
@Plerx2493
I've been building this bot for several months now, but it's my side project so that's not many real hours spent. My bot is fairly simple in its discord part (mostly communication with another API) so migration to 5.0 will be fairly easy for me.

4.4.6-4.4.8 don't have `DiscordClientBuilder`. It seems to be 5.0 feature. I was not able to find a matching equivalent for 4.4.x

**Comment by Plerx2493** [comment]
Oh yeah sry, I forgot that our docs are v5 already. Just take a look at event handlers in c#. V4 uses standard c# patterns for events

**Comment by akiraveliara** [comment]
fixed in v4.4.9
